pkg/controller/stack/stack_controller.go

```diff
@@ -547,6 +525,20 @@ func (sess *reconcileStackSession) SetupPulumiWorkdir(gitAuth *auto.GitAuth) err
 	return nil
 }
 
+// Determine the actual commit information from the working directory (Spec commit etc. is optional).
+func (sess *reconcileStackSession) updateGitRevisionAtWorkingDir() error {
+	gitRepo, err := git.PlainOpen(sess.workdir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve git repository from working directory: %s", sess.workdir)
+	}
+	headRef, err := gitRepo.Head()
+	if err != nil {
+		return errors.Wrapf(err, "failed to determine revision for git repository at %s", sess.workdir)
+	}
+	sess.currentCommit = headRef.Hash().String()
```

Comment:
The way we currently do all this mutation of `sess` through these various methods definitely feels a little brittle - though I recognize you are continuing with the pattern that already largely exists here.  Not sure it's worth it for a one-off here - but I could imagine this function just returning the commit (and not even being a method on the session), just to make the action-at-a-distance aspect a little less bad here.

Reply:
Related - if we *aren't* going to do that - I would rename this to `setCurrentCommit` or something similar to be more clear on what the function does.

Reply:
I agree with your original statement. In general the the mutative changes are a bit heavy here. I went with the approach you outlined in your former comment.